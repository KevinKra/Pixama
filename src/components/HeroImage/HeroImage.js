import React from "react";
import { curatedData } from "../../_assets/curatedHeroData";
import "./HeroImage.scss";

export default function HeroImage() { 
  const divStyle = (image, position) => {
    return {
      backgroundImage: `url(${image})`,
      backgroundPosition: position,
      backgroundSize: "cover",
      backgroundRepeat: "no-repeat"
    };
  };

  const selectFromPool = () => {
    const index = Math.floor(Math.random() * (curatedData.length - 1));
    const movie = curatedData[index];
    return (
      <section className="HeroContent">
        <div
          className="hero-image"
          style={divStyle(
            `https://image.tmdb.org/t/p/original${movie.backdrop_path}`,
            "center top"
          )}
        />
        <div className="opacity-filter" />
        <div className="hero-movie-details">
          <h2>{movie.original_title}</h2>
          <p>{movie.overview}</p>
          <button>Watch</button>
        </div>
        <div className="hero-movie-support">
          <p className="rating" disabled>
            {movie.rating}
          </p>
        </div>
      </section>
    );
  };

  return <React.Fragment>{selectFromPool()}</React.Fragment>;
}
