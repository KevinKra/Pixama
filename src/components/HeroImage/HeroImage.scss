.HeroImage {
  position: relative;
  z-index: -1;
  border: 1px solid red;
  width: 100%;
  height: 85vh;
}

.hero-image {
  position: absolute;
  background-size: cover;
  background-position: center;
  border: 1px solid green;
  height: 100%;
  width: 100%;
  z-index: 1;
}

.hero-text {
  position: absolute;
  z-index: 2;
  color: white;
}

.screen {
  position: absolute;
  z-index: 1;
  background-color: rgba(0, 0, 0, 0.664);
  height: 100%;
  width: 100%;
}
