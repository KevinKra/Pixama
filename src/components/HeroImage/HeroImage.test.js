import HeroImage from "./HeroImage";
import React from "react";
import { shallow } from "enzyme";

describe('HeroImage', () => {
  let wrapper;
  let instance;

  beforeEach(() => {
    wrapper = shallow(<HeroImage />);
    instance = wrapper.instance();
  });

  it ('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
