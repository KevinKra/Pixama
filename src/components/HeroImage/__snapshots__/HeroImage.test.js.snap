// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroImage should match the snapshot 1`] = `
<Fragment>
  <section
    className="HeroContent"
  >
    <div
      className="hero-image"
      style={
        Object {
          "backgroundImage": "url(https://image.tmdb.org/t/p/original/m67smI1IIMmYzCl9axvKNULVKLr.jpg)",
          "backgroundPosition": "center top",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "cover",
        }
      }
    />
    <div
      className="opacity-filter"
    />
    <div
      className="hero-movie-details"
    >
      <h2>
        Toy Story 4
      </h2>
      <p>
        Woody has always been confident about his place in the world and that his priority is taking care of his kid, whether that's Andy or Bonnie. But when Bonnie adds a reluctant new toy called "Forky" to her room, a road trip adventure alongside old and new friends will show Woody how big the world can be for a toy.
      </p>
      <button>
        Watch
      </button>
    </div>
    <div
      className="hero-movie-support"
    >
      <p
        className="rating"
        disabled={true}
      >
        G
      </p>
    </div>
  </section>
</Fragment>
`;
